github_actions/lib/dependabot/github_actions/file_updater.rb

```diff
@@ -65,17 +65,37 @@ def updated_workflow_file_content(file)
             gsub(/@.*+/, "@#{new_req.fetch(:source).fetch(:ref)}")
 
           # Replace the old declaration that's preceded by a non-word character
-          # and followed by a whitespace character (comments) or EOL
+          # and followed by a whitespace character (comments) or EOL.
+          # If the declaration is followed by a comment, attempt to update
+          # any version comments associated with SHA source refs.
           updated_content =
             updated_content.
             gsub(
-              /(?<=\W|"|')#{Regexp.escape(old_declaration)}(?=\s|"|'|$)/,
-              new_declaration
-            )
+              /(?<=\W|"|')#{Regexp.escape(old_declaration)}(?<comment>\s+#.*)?(?=\s|"|'|$)/
+            ) do |s|
+              comment = Regexp.last_match(:comment)
```

Comment:
Something like this might also work
```ruby
s.gsub!(old_declaration, new_declaration)
if (comment = Regexp.last_match(:comment))
  if (updated_comment = updated_version_comment(comment, new_req))
    s.gsub!(comment, updated_comment)
  end
end
```

Reply:
I got this suggestion working and I like it much better. Thanks!